### ISSUE TITLE
NVDA reads skip to main content link before it's visible

### ISSUE THREAD

**Issue by maspychaj-sage** [Sage/carbon#4928]
### Current behaviour

NVDA on focus page reads also skip to main content link which is not visible:
![image](https://user-images.githubusercontent.com/75613789/156528348-8112dffa-63c6-4d77-b9fe-7fc0b4259b36.png)

### Expected behaviour

Skip to main content link should only be read by NVDA when it's visible.

### CodeSandbox or Storybook URL

https://carbon.sage.com/iframe.html?id=link--is-skip-link&args=&viewMode=story

### JIRA Ticket (Sage Only)

SBS-28470

### Carbon Version

104.38.2

### What browsers are you seeing the problem on?

Chrome

### What Operating System are you seeing the problem on?

Windows

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by sianford** [comment]
Confirmed this is an issue in both NVDA and VoiceOver. The 'Skip to main content' link is not on screen, however the screen readers seem to find it and focus on it.
<img width="1792" alt="VoiceOver_Skip_to_main_content_link" src="https://user-images.githubusercontent.com/12828099/160657002-cdab349f-1ab5-4008-8140-b36720c6cda1.png">
<img width="1792" alt="NVDA_Skip_to_main_content_link" src="https://user-images.githubusercontent.com/12828099/160657019-3047682e-a2aa-44d4-85e4-243453a6ac9e.png">

**Comment by jamime** [comment]
@sianford is going to discuss this with @tempertemper.

**Comment by samtjo** [comment]
@sianford @tempertemper - hi both, did we get anywhere in discussion with this one?

**Comment by tempertemper** [comment]
I spoke to @sianford a couple of weeks ago and we suspected it was the method that has been used to hide the skip link. Having had a look at GOV.UK, https://adrianroselli.com/, https://www.scottohara.me/, and one or two other reputable sites with skip links, NVDA behaves the same on them all, reading the skip link on page load.

I've tested the same sites with VoiceOver and JAWS and it doesn't read the skip link. So I'm inclined to think this is a quirk of NVDA and that NVDA users will be accustomed to hearing about the skip link. If we used a different method to hide it that meant it didn't read for NVDA users, we stand to upset this expectation, noisy though it may be.

I'm happy for this issue to be closed as it's a quirk of a particular piece of software that might even be fixed at some point down the line.

**Comment by samtjo** [comment]
Thanks @tempertemper 👍🏻  I'll close this issue then.
